### ISSUE TITLE
Move MD file to other folder

### ISSUE THREAD

**Issue by HaroldLee931** [foambubble/foam#265]
I found that the markdown link graph will not show the link between the MDs in different folder.
Also when I make a new folder under my foam_working path,  the graph can not display the link relationship for the MDs which are located in the new folder.

**Comment by scott-joe** [comment]
Are the files linked using the Wikilink format of `[[filename]]`? The filename format and location are all fairly fragile right now. I've found that I have to ensure every file in the workspace has a unique name, it's downward accessible in the directory tree, and has a `#H1` to give it a useful title. I believe the idea right now is that _all_ markdown files would be in the same directory and the graph would be your file navigation system. That's not how devs typically think about files, so that's a hard sell right now.

**Comment by HaroldLee931** [comment]
> accessible
 OK, I understand. Thanks for your reply.
Seems I need to take care for the filename for now, if I want to make some cross domain(research area) notes.
BTW, this is a really cool tool.

**Comment by scott-joe** [comment]
Right. I've got a weird solution that I'd like to move into the Foam tool itself, but this may help you organize your notes.

https://github.com/scott-joe/foam--workspace is my workspace and https://github.com/scott-joe/foam--bottle--personal is an example of a collection of notes. There's a description of my approach in the workspace repo. Let me know if it's close to what you're trying to do.

**Comment by riccardoferretti** [comment]
There are several improvements in the works that would come with support for LSP, including sounder wikilinks - you can see a demo of the current prototype here: https://youtu.be/ZqLEGGmOw-8

This will also have proper support for directories (theoretically it's already there but as you said it's brittle)

**Comment by scott-joe** [comment]
https://www.youtube.com/watch?v=ZqLEGGmOw-8&feature=youtu.be In this, it looks like some work is being done on how Foam sees files and directories, but the graph is another project. It may require some parallel work or a fork to get it working with sub-directories.

**Comment by scott-joe** [comment]
Subscribed!

**Comment by vojavocni** [comment]
I've noticed that, if I move a file to another folder and the links get broken, I can then open all files that reference the moved file and the links would fix themselves automaticall. I'm wondering if it's possible to have a command in vscode that would fix the links in all files automatically, without having to go and open them?

**Comment by riccardoferretti** [comment]
We are currently adding support for refactoring (file name/remove) - no VS Code command will be required, it will just work

**Comment by vojavocni** [comment]
Awesome! Thanks for the answer! 🙂

**Comment by lllAlexanderlll** [comment]
Hi,
I tried to move the folders "features" and "how-to" from the [template](https://github.com/foambubble/foam-template) into a new folder "foam" and this broke all links. Thus, I assume this feature not yet arrived. What is the current status of this?

**Comment by lllAlexanderlll** [comment]
Thanks for the helpful link!
When I "Foam: Show Graph", foam seems to still know about the link between the moved files:
![image](https://user-images.githubusercontent.com/12560547/176843400-1ce62c66-4358-468b-8ac5-4e0b654ade71.png)
I tried opening  `[[tags-and-tag-explorer]]` from `docs/foam/how-to/get-started-with-vscode.md`, which should point to the moved file `docs/foam/features/tags-and-tag-explorer.md`, which was before moving at `docs/features/tags-and-tag-explorer.md`, which is where vsCode is trying to open the file in a new tab.

**Comment by lllAlexanderlll** [comment]
Now it works magically. I'm not aware that I changed anything. Are there routines in the background which "clean up" the repo from time to time? Sorry that I cannot document what changed the status to "working".
